src/main/java/org/takes/misc/Opt.java

```diff
@@ -49,6 +51,54 @@
     boolean has();
 
     /**
+     * Holder for a possible null element.
+     *
+     * <p>The class is immutable and thread-safe.
+     * @author I. Sokolov (happy.neko@gmail.com)
+     * @version $Id$
+     * @since 0.32
+     * @param <T> Type of item
+     * @todo #608:30min All library methods calls that may return null
+     *  should be wrapped in Opt.Nullable and checked for value presence
+     *  with Opt.has().
+     */
+    final class Nullable<T> implements Opt<T> {
+        /**
+         * Origin.
+         */
+        private final transient T origin;
+        /**
+         * Origin is non-null.
+         */
+        private final transient boolean has;
+
+        /**
+         * Ctor.
+         * @param orgn The possibly-null item to hold
+         */
+        public Nullable(final T orgn) {
+            this.origin = orgn;
+            this.has = orgn != null;
```

Comment:
@Happy-Neko this field is not neccessary, you should check for null in the get method (somehow related http://www.yegor256.com/2016/01/26/defensive-programming.html), you should check for valid values only during retrieval